File: src/brs/web/api/http/handler/GetBalance.java
package brs.web.api.http.handler;

import brs.BurstException;
import brs.web.api.http.common.LegacyDocTag;
import brs.web.api.http.common.Parameters;
import brs.services.ParameterService;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.JSONData;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

public final class GetBalance extends ApiServlet.JsonRequestHandler {

  private final ParameterService parameterService;

  public GetBalance(ParameterService parameterService) {
    super(new LegacyDocTag[]{LegacyDocTag.ACCOUNTS}, Parameters.ACCOUNT_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws BurstException {
    return JSONData.accountBalance(parameterService.getAccount(req));
  }

}

File: src/brs/http/GetBalance.kt
package brs.http

import brs.BurstException
import brs.http.common.Parameters
import brs.services.ParameterService
import com.google.gson.JsonElement

import javax.servlet.http.HttpServletRequest

internal class GetBalance(private val parameterService: ParameterService) : APIServlet.JsonRequestHandler(arrayOf(APITag.ACCOUNTS), Parameters.ACCOUNT_PARAMETER) {

    @Throws(BurstException::class)
    internal override fun processRequest(req: HttpServletRequest): JsonElement {
        return JSONData.accountBalance(parameterService.getAccount(req))
    }

}
